Parse/cloud/main.js

```diff
@@ -41,6 +59,56 @@ Parse.Cloud.beforeSave("VentAudience", async (request) => {
 
 });
 
+Parse.Cloud.beforeSave("GroupDetails", async (request) => {
+  var currentGroup = request.object;
+  var groupName = await request.object.get("groupName");
+
+  request.object.set(
+    "searchName",
+    groupName.toLowerCase()
+  );
+
+});
+
+Parse.Cloud.afterSave("GroupDetails", async (request) => {
+  const SearchItem = Parse.Object.extend("SearchItem");
+  const searchItem = new SearchItem();
+  var currentGroup = request.object;
+  var groupName = await request.object.get("groupName");
+
+  searchItem.save({
+    searchName: groupName.toLowerCase(),
+    group: currentGroup
+  })
+  .then((searchItem) => {
+    // The object was saved successfully.
+  }, (error) => {
+    // The save failed.
+    // error is a Parse.Error with an error code and message.
+  });
+
+});
+
+Parse.Cloud.beforeSave("SearchItem", async (request) => {
+  var group = await request.object.get("group");
+  var user = await request.object.get("user");
+
+  if (group !== undefined) {
+    request.object.set(
+      "groupId",
+      group.id
+    );
+  }
+
+  if (user !== undefined) {
+    request.object.set(
+      "userId",
+      user.id
+    );
+  }
```

Comment:
this is a dangerous pattern, we should do if (group) else if (user) else { log an error}

basically, we should definitely log an error if there is neither a user or a group, maybe even throw an exception so that the entire request fails. You can do something like this

```
...
else {
  throw new Error("Search item doesn't have a user or a group, and must have one");
}
```

it would also be good to confirm above that it doesn't have _both_ a group and a user, which I don't think would make sense, so good to add simple checks like this and throw errors if they happen